x-pack/plugins/security/server/lib/saved_objects_client/secure_saved_objects_client.js

```diff
@@ -91,6 +91,15 @@ export class SecureSavedObjectsClient {
     );
   }
 
+  async incrementCounter(type, id, counterFieldName, options = {}) {
```

Comment:
@kobelb would also appreciate a look at this to make sure I didn't do anything stupid, I only stumbled across this file after my changes to the regular saved_object_client weren't working.

Reply:
When users opt into telemetry, is this being stored in the uiSettings, and if so I assume that read-only users aren't able to opt into it?

Reply:
If a user clicks the toggle in the query bar (what we're tracking with this counter) it stores their preference in localstorage. readonly users would only be prevented from setting the global default.

So I think I see where you're going with this. A readonly user could opt in but the request to increment the counter in .kibana is going to fail if we use their credentials. I've already encapsulated the saved object client in a purpose specific HTTP API endpoint, so I'm guessing we could use the internal kibana user instead. Do you know if there's an instance of the saved object client available that's not request bound?

Reply:
If we're ensuring that they can write to the uiSettings before hand, then we shouldn't run into an  issue here and it's generally preferred to execute actions initiated as end-users in the context of the end-user and not use the internal server user here.

Reply:
Gotcha, I misunderstood earlier. 

> I can't see why it would be a bad idea to use the internal user in this route. The request to ES doesn't use any information provided by the user, the user action simply triggers a pre-made query on the server.

As long as we're only doing so using a dedicated API where all of these conditions are true, we should be alright from a security perspective. This assumes that we're only allowing authenticated users to use the API, which should be the case as whenever security is enabled and we haven't opted the API out of this protection.

To get an instances of the SavedObjectsClient that uses the internal user you can use code similar to the following:
```
const { savedObjects } = server;

const adminCluster = server.plugins.elasticsearch.getCluster('admin');
const { callWithInternalUser } = adminCluster;
const internalRepository = savedObjects.getSavedObjectsRepository(callWithInternalUser);
const savedObjectsClient = new savedObjects.SavedObjectsClient(internalRepository)
```

Reply:
Sweet, that works! 
https://github.com/elastic/kibana/pull/23547/commits/a48856bfb0570c4b96515deab2812d2cc4ff5487